llvm/test/Transforms/ArgumentPromotion/argpromotion-recursion-pr1259.ll

```diff
@@ -0,0 +1,65 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
```

Comment:
This needs a lot more tests.  In particular, tests for the various cases we discussed that should not be transformed.

Reply:
makes sense, I will add testcase for these two cases:

1) check for argument position of the caller and callee.
2) check if the pointer offset is non-zero

lmk, if I need to add anything more.

Reply:
added two more testcases, not sure what more I can add, any suggestions ?

Reply:
You can probably put all the testcases in one file.

In terms of what to test... non-zero offsets, functions containing both recursive and non-recursive calls, a struct with multiple elements.

Reply:
> a struct with multiple elements.

can you help with this ? not sure what is expected.